### PR TITLE
Update discovery to 16.3.4

### PR DESCRIPTION
**Pull request by rabajaj0509**

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
It seems to me, like some build command is missing the `DATABASE_URL=nulldb://nohost` as it's trying to connect to the real database.

File: plugins/ruby-foreman-discovery/debian/rules [hunk 1]
```diff
@@ -16,10 +16,14 @@ build:
 	cp $(PLUGIN).rb /usr/share/foreman/bundler.d/
 	cd /usr/share/foreman && ( \
 		/usr/bin/foreman-ruby /usr/bin/bundle install --local && \
+		/usr/bin/npm install --no-audit --no-optional --unsafe-perm && \
 		/usr/bin/foreman-ruby /usr/bin/bundle exec rake plugin:apipie:cache[$(PLUGIN)] cache_part=resources \
 			OUT=/var/lib/foreman/public/apipie-cache/plugin/$(PLUGIN) RAILS_ENV=development \
 			DATABASE_URL=nulldb://nohost FOREMAN_APIPIE_LANGS=en_US \
```
Comment (rabajaj0509) [inline plugins/ruby-foreman-discovery/debian/rules:22]: @ezr-ondrej on line 22 we already have the `DATABASE_URL=nulldb://nohost`, is this what you were referring to?

File: plugins/ruby-foreman-discovery/debian/rules [hunk 2]
```diff
@@ -16,10 +16,14 @@ build:
 	cp $(PLUGIN).rb /usr/share/foreman/bundler.d/
 	cd /usr/share/foreman && ( \
 		/usr/bin/foreman-ruby /usr/bin/bundle install --local && \
+		/usr/bin/npm install --no-audit --no-optional --unsafe-perm && \
```
Comment (evgeni) [inline plugins/ruby-foreman-discovery/debian/rules:19]: I think you'll also need:
```suggestion
		/usr/bin/npm install --no-audit --no-optional --unsafe-perm && \
		/usr/bin/foreman-ruby /usr/bin/bundle exec rake plugin:assets:precompile[$(PLUGIN)] RAILS_ENV=production DATABASE_URL=nulldb://nohost && \
```

File: plugins/ruby-foreman-discovery/debian/control [hunk 3]
```diff
@@ -8,6 +8,6 @@ Homepage: https://github.com/theforeman/foreman_discovery
 
 Package: ruby-foreman-discovery
 Architecture: all
-Depends: ${misc:Depends}, bundler, foreman (>= 1.24.0~rc1)
+Depends: ${misc:Depends}, bundler, foreman-assets(>= 2.0.0~rc1), foreman (>= 1.24.0~rc1)
```
Comment (evgeni) [inline plugins/ruby-foreman-discovery/debian/control:11]: assets needs to be in BuildDepends -- line 5

and the versions of assets and foreman should match ;)

Comment (rabajaj0509) [reply plugins/ruby-foreman-discovery/debian/control:11]: ah! my bad.

**Comment by evgeni** [comment]
seems like it built now.

**Comment by lzap** [comment]
Does it also work, @rabajaj0509? We will be backporting this so it would be good to confirm prior doing so.

**Comment by rabajaj0509** [comment]
@lzap nope, I have only followed few simillar PRs (foreman-tasks and foreman-ansible plugin) to get to this, let me setup the env and test it properly and get back to you on the PR.

**Comment by lzap** [comment]
@evgeni do you know why @rabajaj0509 don't have directory in https://stagingdeb.theforeman.org/dists/buster/ ?

**Comment by evgeni** [comment]
Because it's a plugin. http://stagingdeb.theforeman.org/pool/plugins/rabajaj0509/ exists. See docs how to use plugin staging at http://stagingdeb.theforeman.org/

**Comment by rabajaj0509** [comment]
I have tested this PR on the debian nightly machine, when I install the package with a `--force-overwrite` option everything works fine, discovery is loaded and the react page looks good.
discoveredHosts


Although, when I try to install the package withou the --force-overwrite option:
`dpkg -i ruby-foreman-discovery_9999-plugin+scratchbuild+20201123102820_all.deb`
```
Unpacking ruby-foreman-discovery (9999-plugin+scratchbuild+20201123102820) ...
dpkg: error processing archive /home/vagrant/ruby-foreman-discovery_9999-plugin+scratchbuild+20201123102820_all.deb (--unpack):
 trying to overwrite '/var/lib/foreman/public/webpack/fonts/RedHatDisplay-Bold.bundle-v6.0.0-production-b72e57f15dfaf4157a980a740627cc3e.eot', which is also in package foreman 9999-buster+scratchbuild+20201119192508
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Errors were encountered while processing:
 /home/vagrant/ruby-foreman-discovery_9999-plugin+scratchbuild+20201123102820_all.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Let me know your thoughts @lzap @evgeni

**Comment by evgeni** [comment]
ah, interesting, yeah I think we also exclude that fonts dir in the RPM packages somewhere…

**Comment by evgeni** [comment]
ah, here:

https://github.com/theforeman/foreman-packaging/blob/c5b27197ddeb21227dbdf4843b6a12f2fa0f6512/packages/foreman/foreman/foreman.spec#L904-L907

these are the things we clean up

**Comment by lzap** [comment]
Error:

```
00:12:18.044 debian/rules:32: recipe for target 'binary' failed
00:12:18.044 make: *** [binary] Error 25
```

**Comment by evgeni** [comment]
The real error is in the lines above:

```
13:07:08 dh_install: Cannot find (any matches for) "webpack/foreman-discovery" (tried in ., debian/tmp)
13:07:08 
13:07:08 dh_install: ruby-foreman-discovery missing files: webpack/foreman-discovery
13:07:08 dh_install: Cannot find (any matches for) "assets" (tried in ., debian/tmp)
13:07:08 
13:07:08 dh_install: ruby-foreman-discovery missing files: assets
13:07:08 dh_install: missing files, aborting
```

File: plugins/ruby-foreman-discovery/debian/install [hunk 4]
```diff
@@ -1,3 +1,5 @@
-foreman_discovery.rb usr/share/foreman/bundler.d
-cache                usr/share/foreman/vendor
-apipie-cache/foreman_discovery    var/lib/foreman/public/apipie-cache/plugin
+foreman_discovery.rb             usr/share/foreman/bundler.d
+cache                            usr/share/foreman/vendor
+webpack/foreman-discovery        var/lib/foreman/public/webpack
```
Comment (evgeni) [inline plugins/ruby-foreman-discovery/debian/install:3]:
```suggestion
webpack/foreman_discovery        var/lib/foreman/public/webpack
```

File: plugins/ruby-foreman-discovery/debian/install [hunk 5]
```diff
@@ -1,3 +1,5 @@
...
+apipie-cache/foreman_discovery   var/lib/foreman/public/apipie-cache/plugin
+assets                           var/lib/foreman/public
```
Comment (evgeni) [inline plugins/ruby-foreman-discovery/debian/install:5]: you don't have any non-webpack assets?
```suggestion
```

Comment (rabajaj0509) [reply plugins/ruby-foreman-discovery/debian/install]: makes sense :) thanks.

File: plugins/ruby-foreman-discovery/debian/rules [hunk 6]
```diff
@@ -16,10 +16,15 @@ build:
...
+		/usr/bin/foreman-ruby /usr/bin/bundle exec rake plugin:assets:precompile[$(PLUGIN)] RAILS_ENV=production DATABASE_URL=nulldb://nohost && \
 		/usr/bin/foreman-ruby /usr/bin/bundle exec rake plugin:apipie:cache[$(PLUGIN)] cache_part=resources \
 			OUT=/var/lib/foreman/public/apipie-cache/plugin/$(PLUGIN) RAILS_ENV=development \
 			DATABASE_URL=nulldb://nohost FOREMAN_APIPIE_LANGS=en_US \
 	)
+	GEM_PATH=$$(cd /usr/share/foreman && /usr/bin/foreman-ruby /usr/bin/bundle show $(PLUGIN)) && \
+	cp -rp $${GEM_PATH}/public/assets/$(PLUGIN) ./ && \
```
Comment (evgeni) [inline plugins/ruby-foreman-discovery/debian/rules:26]:
```suggestion
	cp -rp $${GEM_PATH}/public/assets/ ./ && \
```

File: plugins/ruby-foreman-discovery/debian/install [hunk 7]
```diff
@@ -1,3 +1,4 @@
-foreman_discovery.rb usr/share/foreman/bundler.d
-cache                usr/share/foreman/vendor
-apipie-cache/foreman_discovery    var/lib/foreman/public/apipie-cache/plugin
+foreman_discovery.rb             usr/share/foreman/bundler.d
+cache                            usr/share/foreman/vendor
+webpack/foreman_discovery        var/lib/foreman/public/webpack
+apipie-cache/foreman_discovery   var/lib/foreman/public/apipie-cache/plugin
```
Comment (evgeni) [inline plugins/ruby-foreman-discovery/debian/install:4]:
```suggestion
apipie-cache/foreman_discovery   var/lib/foreman/public/apipie-cache/plugin
assets                           var/lib/foreman/public
```

the below `cp` change makes `assets` exist XD

**Review by evgeni — APPROVED** [review]

**Comment by lzap** [comment]
Thanks all. 2.3 CP: https://github.com/theforeman/foreman-packaging/pull/6042

**Pull request merged**